targets/TARGET_Cypress/TARGET_PSOC6/common/cy_network_buffer.c

```diff
@@ -23,7 +23,7 @@
 * limitations under the License.
 *******************************************************************************/
 
-#if defined(CYBSP_WIFI_CAPABLE)
+#if MBED_CONF_CY_PSCOC6_WHD_PRESENT
```

Comment:
It is not valid to add Mbed-specific macros to cy_network_buffer.c and cybsp_wifi.c, this is common code shared across multiple ecosystems:

https://github.com/cypresssemiconductorco/wifi-mw-core/blob/master/lwip-whd-port/cy_network_buffer.c

https://github.com/cypresssemiconductorco/amazon-freertos/blob/master/vendors/cypress/psoc6/common/cy_network_buffer.c

Reply:
The problem is that both modules pull in WHD which depends on RTOS functionality which is not present with the bare metal profile.
More info [here](https://github.com/ARMmbed/mbed-os/pull/12438#discussion_r379834194).

Reply:
I have made changes to address the comment [here](https://github.com/ARMmbed/mbed-os/pull/12438#issuecomment-586703133).